### PR TITLE
clean up pet service accounts

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/AP-210

Google has told us the recreating service accounts can lead to unpredictable behavior. They suggest removing a service account from all groups before deleting and recreating. This PR deletes pets for all users in a billing project when the project is unregistered (unregistering projects only happens in tests). The sam code to delete a pet disables it (removes it from proxy group) so this _should_ do as google suggests.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]

**Review by marctalbott — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 1]
```diff
@@ -194,13 +196,23 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
   def unregisterBillingProject(projectName: RawlsBillingProjectName, ownerInfo: Map[String, String]): Future[PerRequestMessage] = {
     val ownerUserInfo = UserInfo(RawlsUserEmail(ownerInfo("newOwnerEmail")), OAuth2BearerToken(ownerInfo("newOwnerToken")), 3600, RawlsUserSubjectId("0"))
     for {
+      projectUsers <- samDAO.listAllResourceMemberIds(SamResourceTypeNames.billingProject, projectName.value, ownerUserInfo)
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:199]: I take it this returns users (proxies), not users + pets?

**Comment by dvoet** [comment]
jenkins retest - oidc failure

**Review by helgridly — APPROVED** [review]
How are we recreating pets in tests? I thought tests used fresh projects every time?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala (hunk 1 above)
Comment (dvoet) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:199]: just users

**Comment by dvoet** [comment]
tests use projects from gpalloc which aren't really as fresh as a newly created project. This situation happens during the run of an automation suite:
* test 1 gets a project from gpalloc, does stuff that creates a pet SA
* test 1 releases project, gpalloc scrubs the project (including pet SA)
* test 2 gets a project from gpalloc and it happens to be the same one that was used in 1
* test 2 does something which creates a new pet SA but with the same name as in test 1

**Comment by helgridly** [comment]
oh right. thanks, the specifics slipped off my exceptionally smooth brain. and we don't tidy up proxy and auth groups when we're finished.

**Pull request merged**